api.yaml

```diff
@@ -89,6 +89,75 @@ paths:
             #! if eq .Response.Body.errorType "InvalidInput" !# 400
             #! else !# 500
             #! end !#, "bodyPassthrough": true}'
+  /cloud:
+    get:
+      summary: "Retrieve a cloud asset"
+      parameters:
+        - name: "ipAddress"
```

Comment:
We need to be careful with this API. Any query param may be given more than once but we don't define a behavior for this. Additionally, it's not clear from the specification if these conditions are combined with `OR` or `AND` or something else.

I'm also hesitant to offer one API that makes all attributes queryable. This quickly degenerates into "implement an API that allows consumers to run arbitrary SQL statements in the database". I'd much prefer to start with a very limited set of operations targeted at attribution for specific details of the asset. This might be, for example, an API of `/cloud/ip/{ipAddress}?time=XXX`. I'm open to different ways of shaping the API but I do think we should start with heavy restrictions on what we support in terms of search complexity.

Reply:
I've updated the API to match the more targeted `/cloud/ip/{ipAddress}?time=XXX` style